runtime/autoload/dist/ft.vim

```diff
@@ -1191,6 +1191,32 @@ export def FTdat()
   endif
 enddef
 
+# Determine if a *.l file is Lex or PicoLisp (default to Lex)
+export def FTl()
+  if exists("g:filetype_l")
+    exe "setf " .. g:filetype_l
+    return
+  endif
+  for i in range(1, line("$"))
```

Comment:
Many filetype detection functions have a limit to the number of lines to check, like using `range(1, min([line("$"), 100]))`